baybe/utils/chemistry.py

```diff
@@ -72,158 +72,117 @@ def name_to_smiles(name: str) -> str:
 
 @lru_cache(maxsize=None)
 @_disk_cache
-def _smiles_to_mordred_features(smiles: str) -> np.ndarray:
-    """Memory- and disk-cached computation of Mordred descriptors.
-
-    Args:
-        smiles: SMILES string.
-
-    Returns:
-        Mordred descriptors for the given smiles string.
-    """
-    try:
-        return np.asarray(
-            _mordred_calculator(Chem.MolFromSmiles(smiles)).fill_missing()
-        )
-    except Exception:
-        return np.full(len(_mordred_calculator.descriptors), np.nan)
-
-
-def smiles_to_mordred_features(
-    smiles_list: list[str],
-    prefix: str = "",
-    dropna: bool = True,
-) -> pd.DataFrame:
-    """Compute Mordred chemical descriptors for a list of SMILES strings.
+def _molecule_to_fingerprint_features(
```

Comment:
I still think this is confusing: This is called `_molecule`, but also accepts `SMILES`, and I still think that it is weird to have this one-liner as a single function. Is it really necessary to have this as an individual function, given that it also seems to be called in exactly one position only (if I did not overlook anything)?
So I'd be for "Let's delete this function and save multiple lines of code and just write `encoder.transform([molecule])` at the one position where this is actually used"

Reply:
@AVHopp: you are overlooking the caching decorators, which are the critical part here. But regarding naming, we can change things. But here @Hrovatin and @Scienfitz need to say what is the appropriate term to describe the input in the most general way, i.e. whether it is "molecule" or "substance" or whatever – I can't tell

Reply:
I do not understand the issue you have, is it because of the type? I guess youd accept to call `Chem.Mol` a `molecule` right? So the issue is with `str` ? Not sure how to solve that, SMILES are just a strin based representation of `molecule`s

perhaps `_smiles_or_mol_to_features` ??

Reply:
I think this makes it overly complicated, especially given that this is a private function anyway. Because the more critical part of @AVHopp' question (i.e. the caching) has been answered, I suggest we keep the rest as is because – and I agree with @Scienfitz here – both SMILES and `chem.Mol` are just two representations of a molecule. Or to put it differently: we are not passing an "actual" molecule to the function... I guess no one of us has some form of interface on their PC that allows to pass an actual substance to the CPU 😅 So in that sense, it's obvious that whatever the function takes is just a representation of a molecule, and both types satisfy this requirement. Resolving 🙃